QEfficient/generation/text_generation_inference.py

```diff
@@ -171,11 +172,11 @@ def cloud_ai_100_exec_kv_helper(
     inputs = tokenizer(prompt, return_tensors="np", padding=True)
     position_ids_update = inputs["attention_mask"].sum(1, keepdims=True)
     if generation_len is None:
-        generation_len = ctx_len
+        assert generation_len > 0, "generation length should be greater than zero, please pass valid generation length"
```

Comment:
Here, we should be using generation_len = ctx_len - input_len

Reply:
The logic is incorrect here. This assertion needs to be done `if generation_len is not None`, but it's done `if generation_len is None`.